Map Appsetting to flag names.

arijitAD opened the pull request:
This PR maps Golang flags to `Appsetting` struct.
For supporting configs for multiple ports, I was wondering to have `[]Appsetting` and have config name inside it that identifies the config and flags will be updated for that config only. There is always a default config. If the config name is not supplied then flags are updated in default config.
I was initially trying to support multiple configs in single struct but couldn't come up with an elegant approach. I am open to comments. 
Even though the current approach will have redundant values I chose it for simplicity.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/buger/goreplay?pullRequest=792) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/buger/goreplay?pullRequest=792) before we can accept your contribution.<br/><hr/>**Rini Tikarya** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/buger/goreplay?pullRequest=792) it.</sub>

slimus commented on elasticsearch.go:
```diff
@@ -15,7 +15,7 @@ import (
 
 func (e *ESUriErorr) Error() string {
-	return "Wrong ElasticSearch URL format. Expected to be: scheme://host/index_name"
+	return "Wrong ElasticSearch URL format. Expected to be: scheme://Host/index_name"
```
host

slimus commented on elasticsearch.go:
```diff
@@ -55,7 +55,7 @@ type ESRequestResponse struct {
 // Parse ElasticSearch URI
 //
-// Proper format is: scheme://[userinfo@]host/index_name
+// Proper format is: scheme://[userinfo@]Host/index_name
```
host

slimus commented on elasticsearch_test.go:
```diff
@@ -32,7 +32,7 @@ func assertNoError(returnedError error, t *testing.T) {
 }
 
-// Argument host:port/index_name
+// Argument Host:port/index_name
```
everywhere in comments and unexported structs we should revert to `host`

slimus commented on gor.go:
```diff
@@ -41,7 +41,7 @@ func main() {
 	// 	if r := recover(); r != nil {
 	// 		fmt.Printf("PANIC: pkg: %v %s \n", r, debug.Stack())
-	// 	}
+	// 	}̦
```
revert dot

slimus commented on http_client_test.go:
```diff
@@ -484,7 +484,7 @@ func TestHTTPClientErrors(t *testing.T) {
 	if resp, err := client.Send(req); err != nil {
 		if s := proto.Status(resp); !bytes.Equal(s, []byte("521")) {
-			t.Error("Should return status 521 for no such host, instead:", string(s))
+			t.Error("Should return status 521 for no such Host, instead:", string(s))
```
host

slimus commented on kafka.go:
```diff
@@ -10,12 +10,21 @@ import (
 
 // KafkaConfig should contains required information to
```
fix comment for exported struct

slimus commented on output_binary.go:
```diff
@@ -8,11 +8,11 @@ import (
-	TrackResponses bool
+	workers        int           `json:"output-binary-workers"`
+	Timeout        time.Duration `json:"output-binary-timeout"`
+	BufferSize     int           `json:"output-tcp-response-buffer"`
```
```suggestion
	BufferSize     int           `json:"output-binary-tcp-response-buffer-size"`
```

arijitAD replied:
I think the current name is fine, suggested name is too long for flag value.

slimus commented on output_http.go:
```diff
@@ -62,28 +62,28 @@ type response struct {
-	OriginalHost bool
-	BufferSize   int
+	Timeout      time.Duration `json:"output-http-timeout"`
+	OriginalHost bool          `json:"http-original-Host"`
```
```suggestion
	OriginalHost bool          `json:"output-http-original-host"`
```

slimus commented on output_http.go:
```diff
@@ -62,28 +62,28 @@ type response struct {
 // HTTPOutputConfig struct for holding http output configuration
 type HTTPOutputConfig struct {
-	redirectLimit int
+	redirectLimit int `json:"output-http-redirects"`
```
```suggestion
	redirectLimit int `json:"output-http-redirect-limit"`
```

slimus commented on output_http_test.go:
```diff
@@ -85,16 +85,16 @@ func TestHTTPOutputKeepOriginalHost(t *testing.T) {
 
 	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		if req.Host != "custom-host.com" {
+		if req.Host != "custom-Host.com" {
```
host

slimus commented on output_http_test.go:
```diff
@@ -85,16 +85,16 @@ func TestHTTPOutputKeepOriginalHost(t *testing.T) {
 
-	headers := HTTPHeaders{HTTPHeader{"Host", "custom-host.com"}}
-	Settings.modifierConfig = HTTPModifierConfig{headers: headers}
+	headers := HTTPHeaders{HTTPHeader{"Host", "custom-Host.com"}}
```
host

slimus commented on settings.go:
```diff
@@ -29,64 +29,64 @@ func (h *MultiOption) Set(value string) error {
+	OutputStdout bool        `json:"output-stdout"`
+	OutputNull   bool        `json:"output-null"`
+
+	InputTCP        MultiOption     `json:"input-tcp"`
+	InputTCPConfig  TCPInputConfig  `json:"input-tcp"`
```
two keys have input-tcp json key, is it correct?

arijitAD replied:
Fixed it.

arijitAD commented:
@slimus Changed the settings to exported because in the next PR I will be reading them from a JSON file and directly parse it.
@urbanishimwe @slimus Addressed the comments.
